### ISSUE TITLE
Replace temporary error with separate error type

### ISSUE THREAD

**Issue by wenkokke** [vehicle-lang/vehicle#587]
The following values...
```hs
temporaryUnsupportedAlternatingQuantifiers
temporaryUnsupportedNonLinearConstraint
```
...should be replaced with different constructors without any fields, such as...
```hs
  ...
  | BacktrackFromUnsupportedAlternatingQuantifiers
  | BacktrackFromUnsupportedNonLinearConstraint
```
Perhaps these constructors should even be in a different error type, so that we can statically check if we're always catching them.

**Comment by MatthewDaggitt** [comment]
Yup agreed. It's just difficult to get the monads to align, as we need to throw both the permanent errors and the temporary errors, not to mention we have a lot of helper error functions that assume we're in the permanent error monad.
